es6/animations.js

```diff
@@ -5,180 +5,176 @@ import {
     checkIsValidAnimation,
     checkIsValidSequence,
     EasingFunctions,
+    immutify,
+    computeAnimatedState,
 } from './util.js'
 
+import isEqual from 'lodash.isequal'
 
-const unit_tick = ({start_time, end_time, duration, start_state, end_state, amt, curve='linear', unit=null}, key) => {
-    var {start_state, amt, end_state} = checked_animation_amt({start_state, end_state, amt, key})
-    var {start_time, end_time, duration} = checked_animation_duration({start_time, end_time, duration})
+const tick_func = ({duration, start_state, delta_state,
+                    curve='linear', unit=null}) => {
 
     const curve_func = EasingFunctions[curve]
 
-    return (delta) => {
-        let new_state
-        // These over-boundary cases happen because we need to render
-        // an extra frame before/after start/end times to finish off
-        // animations who's (durations) % (frame rate) != 0.
-        if (delta < 0) {
-            new_state = start_state
-        } else if (delta >= duration) {
-            new_state = end_state
-        } else {
-            // tick progression function, the core math at the heart of animation
-            new_state = start_state + curve_func(delta/duration)*amt
-        }
-        return unit ? `${new_state}${unit}` : new_state
+    return (time_elapsed) => {
+        const bounded_time = Math.min(Math.max(time_elapsed, 0), duration)
```

Comment:
You need the old branching logic to make `AnimateCSS` work properly, bounding the time is not enough.